### PR TITLE
Add realm to the workspace api.

### PR DESCRIPTION
**Pull request by rushtong**
This PR for GAWB-285 merely adds a realm to the API but does nothing with it. 
Implementation of the realm semantics with respect to ACLs is for GAWB-289. 
This task is necessary for UI work to proceed without causing errors.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by rushtong** [comment]
Note to self - need to update the swagger api.

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala [hunk 1]
```diff
@@ -52,6 +53,7 @@ case class WorkspaceRequest (
 case class Workspace(
                       namespace: String,
                       name: String,
+                      realm: Option[String] = None,
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: This should probably be a RawlsGroupRef, not String

Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: I don't think we should give it a default value

**Comment by dvoet** [comment]
I would like to see this persisted to the database as part of this PR. Then the TODOs can be removed.

**Comment by rushtong** [comment]
Gotcha. Even if the realm is persisted, the realm ACL handling still needs to be implemented so the TODOs are still required. They should be addressed in GAWB-289.

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala (hunk 1 above)
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: Not sure I follow the default value comment. This is an optional field, so we're not giving it a default value.

Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala]: you say 
realm: Option[String] = None,

None is the default

instead it should be
realm: Option[String],

**Comment by dvoet** [comment]
We generally do not commit TODOs to develop. Anything that you would make a TODO for should either be resolved or entered into JIRA.

File: src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphWorkspaceDAOSpec.scala [hunk 2]
```diff
@@ -38,6 +39,61 @@ class GraphWorkspaceDAOSpec extends FlatSpec with Matchers with OrientDbTestFixt
...
+        createdDate = testDate,
+        lastModified = testDate,
+        createdBy = "Secret User",
+        attributes = Map("workspace_attrib" -> AttributeString("foo")),
+        accessLevels = Map.empty
+      )
+
+      workspaceDAO.save(secretWorkspace, txn)
+      // now check explicitly that the vertex exists. note that this will fail if our reserved keywords change.
+      assert {
```
Comment (dvoet) [inline src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphWorkspaceDAOSpec.scala]: I would actually load it by calling workspaceDAO.findById and make assert that what is loaded is the same as what was saved.

**Comment by dvoet** [comment]
:+1: after simple test fix, squash, rebase then merge to develop

**Comment by jmthibault79** [comment]
@rushtong After squash, rebase, push, test: please create a new branch **realm** from this branch instead of merging to develop.

**Comment by rushtong** [comment]
Closing this PR and will re-submit against the realm branch

**Pull request closed without merging**